cloudbuild/python-scripts/parse_logs.py

```diff
@@ -0,0 +1,507 @@
+      metric_string: The string to be found
+      end_of_metric_string: Delimiter at the end of the value.
+      query: Query string [if any] that was provided to the connector.
+
+    Returns:
+    Tuple [Metric Value (String), Is Query Result Present (Boolean)]
+      Metric value: which is the sum of metric values obtained from a file.
+       -1 in case metric is not seen in the file.
+      Is Query Result Present: Boolean value. True incase query result
+        is present in the file, False if not.
+    """
+    # Obtain the yarn logs as a string from the GCS bucket.
+    logs_as_string = ''
+    is_query_result_present = False
+    metric_value = -1
+    try:
+        storage_client = storage.Client()
+        bucket = storage_client.get_bucket(cluster_temp_bucket)
+        blob = bucket.blob(gcs_log_object)
+        logs_as_string = blob.download_as_text(encoding='latin-1')
+    except Exception as e:
+        logging.warning(
+            f'[Log: parse_logs WARNING] File {gcs_log_object} Not'
+            f' Found.\nError: {e}'
+        )
+        # Return in case the file was not found is not found.
+        return metric_value, is_query_result_present
+
+    if query:
+        # In case query has been set:
+        # If for any of the record, filter condition is not met then throw an error.
+        is_query_result_present = check_query_correctness(
+            gcs_log_object, logs_as_string
+        )
+
+    # Update the metric value to the actual value.
+    if metric_string in logs_as_string:
+        # If metric string is present in the logs, extract the value.
+        metric_value = extract_metric(
+            logs_as_string, metric_string, end_of_metric_string
+        )
+
+    # If not return the default value (-1) indicating metric not found.
+    return metric_value, is_query_result_present
+
+
+def get_logs_pattern(client_project_name, region, job_id):
+    """Method to return dataproc job state and yarn logs path.
+
+    Args:
+        client_project_name: Project ID of the GCP project that contains the
+          cluster.
+        region: region in which the cluster runs.
+        job_id: JOB_Id of the dataproc job.
+
+    Returns:
+        The path pattern inside the temp bucket in which logs are found.
+
+    Raises:
+        RuntimeError: In case the Dataproc job fails
+    """
+    # Create a client.
+    client = dataproc_v1.JobControllerClient(
+        client_options={'api_endpoint': f'{region}-dataproc.googleapis.com:443'}
+    )
+
+    # Initialize request argument(s).
+    request = dataproc_v1.GetJobRequest(
+        project_id=client_project_name,
+        region=region,
+        job_id=job_id,
+    )
+
+    # Make the request.
+    response = client.get_job(request=request)
+    state = response.status.state.name
+
+    if state == 'ERROR':
+        raise RuntimeError(
+            f'[Log: parse_logs ERROR] Dataproc Job with JOB ID: "{job_id}" failed'
```

Comment:
This is not a log statement, but it contains `[Log: parse_logs ERROR]` in it. Is this intentionally being done? I see this in other places in this file too.

Reply:
Fixed.